tests/unit/cluster.tcl

```diff
@@ -264,13 +264,39 @@ test {Migrate the last slot away from a node using redis-cli} {
         assert_equal OK [$newnode_r CLUSTER SETSLOT $slot NODE $newnode_id]
         assert_equal OK [$owner_r CLUSTER SETSLOT $slot NODE $newnode_id]
 
+        # Using --cluster check make sure we won't get `Not all slots are covered by nodes`.
+        # Wait for the cluster to become stable make sure the cluster is up during MIGRATE.
+        wait_for_condition 1000 50 {
+            [catch {exec src/redis-cli --cluster check 127.0.0.1:[srv 0 port]}] == 0 &&
+            [catch {exec src/redis-cli --cluster check 127.0.0.1:[srv -1 port]}] == 0 &&
+            [catch {exec src/redis-cli --cluster check 127.0.0.1:[srv -2 port]}] == 0 &&
+            [catch {exec src/redis-cli --cluster check 127.0.0.1:[srv -3 port]}] == 0 &&
+            [csi 0 cluster_state] eq {ok} &&
```

Comment:
Do we need `cluster_state` here? Isn't `--cluster check` enough to make sure all nodes agree that all slots are covered?

Reply:
Cluster check just verifies the the config state is self-consistent, waiting for cluster_state to be okay is an independent check that all the nodes actually believe each other are healthy. So both make sense.